protob/messages.proto

```diff
@@ -912,6 +922,102 @@ message CosiSignature {
 	optional bytes signature = 1;		// Signature
 }
 
+///////////////////
+// Iota messages //
+///////////////////
+
+/**
+ * Request: Show seed on the device
+ * @next Success
+ * @next Failure
+ */
+message IotaShowSeed {
+}
+
+/**
+ * Request: Get address counter
+ * @next IotaGetAddressCounter
+ */
+message IotaGetAddressCounter {
```

Comment:
It should be the role of the wallet to store this information, not the device. Unless there is something I am missing.

Reply:
The problem is, when you switch to another computer, it might take a very very long time to log in. Every public address takes 12 seconds to generate. For every transaction ever done, the trezor needs to generate an address and check if there is balance on that address. For a small sacrifice in memory, this process can be much smoother.

Reply:
How can the user restore a new device then? It would take far too long.

Reply:
The address counter is known to the computer and can be saved. It can be restored with IotaSetAddressCounter. If the device is completely lost and the address counter is unknown, it takes up to a few minutes to restore the device, depending on how many transactions have been done with it. It basically then has to generate every single address from index 0 and check against the network if it has a balance.

Reply:
Does the counter have to be sequential (1, 2, 3) or is monotonic (1, 12, 250, 100382) OK?

Reply:
Is there no way to improve the performance of address generation?

Reply:
If the counter is sequential, it is in line with current wallet implementations. Basically it can be anything as long as no numbers are re-used. Sequential also makes recovery faster, compared to monotonic.

 Address generation cannot be done faster. It has to do with the completely different cryptography used, compared to other coins.

Reply:
> Every public address takes 12 seconds to generate. 

12 seconds on which spec? Desktop PC?

Reply:
It's on the TREZOR. Desktop can derive it, if you expose the private key :-p But that defeats the whole use of a TREZOR.

Reply:
That's really unfortunate for HW wallet usecase that you cannot derive the next address without the knowledge of the private key :-/

Reply:
I'm not sure what makes you think otherwise, but this is exactly what I've done already. The entire post-snapshot address re-generation is not required at all.

Reply:
Ok, sorry, I didn't realise that was what you have done already, I must have misunderstood the previous comments. Sorry for my long post.